schema/materials_category/schemas_categorization.yaml

```diff
@@ -0,0 +1,223 @@
+#
+# `file:` - a final JSON file
+# ` - key` - file content key
+# ` - key?` - optional file content key
+# ` - key = ` - file content assignment to types of values
+# `[value1, value2]` - list of values "value1 or value2"
+# `[value, ...]` - list of multiple values "value"
+# `( x | y | z ){1,2}` - a set of values "1 or 2 of any of the three"
+#
+
+material: # already defined
+
+primitive:
+  base:
+    crystal: material
+    crystal_site:
+      # created by reference to some crystal, can be populated with atom, vacancy, void, can be merged with a crystal of the same lattice
+      - host
+      - coordinate
+      - id? # site id in the host crystal
+    atom:
+      - chemical_element
+    vacuum:
+      # is applied along the specified direction, adds vacuum "slab", breaks periodicity in that direction, lowers dimensionality
+      - direction
+      - size
+    void:
+      # can be placed into a crystal centered around specified crystal_site, all atoms inside are removed
+      - center_coordinate
+      - shape
+
+  modifications:
+    strain:
+      # can be applied to a crystal, modifies the lattice parameters and preserves basis in crystal coordinates
+      - matrix
+    repeat:
+      - integer_matrix
+    perturb:
+      # changes coordinates of atoms by applying the function, intended for local displacements
+      - function = ∂r -> f(x,y,z)
+      - is_isometric = bool  # distance preservation -- isometric transformation -- only makes sense in 2D and 1D thin structures
+    fold:
+      # maps 2D flat plane to a curved surface in 3D space
+      - function = f(x, y) -> f(x, y, z)
+      - is_isometric = bool
+  combinations:
+    stack_component:
+      - structure = (material | vacuum)
+    stack:
+      - stack_components = [ stack_component, ... ] # vacuum is added between to add a gap/distance between components
+      - direction = (x | y | z){1}
+    merge:
+      - merge_structures = [ material ]
+      - method = ("adding" | "replacing" | "yielding" | "smart"){1}
+
+pristine_structures:
+  three-dimensional:
+    ideal_crystal: ref:/primitive/base/crystal
+    repeated:
+      supercell:
+        - crystal = ref:/primitive/base/crystal
+        - matrix
+    strained:
+      strained_uniform:
+        - crystal = ref:/primitive/base/crystal
+        - strain_percentage
+      strained_non-uniform:
+        - crystal = ref:/primitive/base/crystal
+        - strain_matrix
+
+  two-dimensional:
+    crystal_lattice_planes:
+      - crystal = material
+      - miller_indices = [ h, k, l ]
+    (termination): # not a configuration, but a data-model
+      - stoichiometry # e.g. SrO, TiO, etc.
+      - symmetry #P6/mmm, P3m1, etc.
+    unique_atomic_layers:
```

Comment:
maybe `atomic_layers_unique` `atomic_layers_unique_repeated` and `atomic_layers`